File: admin-ui/lib/api/types/life-group-analytics.ts
export interface LifeGroupAttendancePoint {
  period: string
  meetings_count: number
  present_count: number
  members_count: number
  attendance_rate: number
}

export interface LifeGroupAttendanceResponse {
  granularity: "month" | "meeting"
  year: number
  month?: number
  rows: LifeGroupAttendancePoint[]
}

export interface LifeGroupAttendanceQuery {
  year?: number
  month?: number
  life_group_id?: number
  granularity?: "month" | "meeting"
}

export interface LifeGroupDistributionBucket {
  label: string
  count: number
}

export interface LifeGroupDistributionResponse {
  by_day: LifeGroupDistributionBucket[]
  by_hour: LifeGroupDistributionBucket[]
  by_neighborhood: LifeGroupDistributionBucket[]
  by_city: LifeGroupDistributionBucket[]
}

export interface LifeGroupDistributionQuery {
  life_group_id?: number
}

File: kmp-mobile/shared/src/commonMain/kotlin/br/church/paz/shared/domain/model/LifeGroupAnalytics.kt
package br.church.paz.shared.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class LifeGroupAttendancePoint(
    val period: String,
    val meetingsCount: Int,
    val presentCount: Int,
    val membersCount: Int,
    val attendanceRate: Double,
)

@Serializable
data class LifeGroupAttendanceAnalytics(
    val granularity: String,
    val year: Int,
    val month: Int?,
    val rows: List<LifeGroupAttendancePoint>,
)

@Serializable
data class LifeGroupDistributionBucket(
    val label: String,
    val count: Int,
)

@Serializable
data class LifeGroupDistributionAnalytics(
    val byDay: List<LifeGroupDistributionBucket>,
    val byHour: List<LifeGroupDistributionBucket>,
    val byNeighborhood: List<LifeGroupDistributionBucket>,
    val byCity: List<LifeGroupDistributionBucket>,
)
